components/director/internal/model/eventing.go

```diff
@@ -1,29 +1,80 @@
 package model
 
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
 type EventingConfiguration struct {
-	DefaultURL string
+	DefaultURL url.URL
 }
 
 type RuntimeEventingConfiguration struct {
 	EventingConfiguration
 }
 
-func NewRuntimeEventingConfiguration(url string) *RuntimeEventingConfiguration {
+func NewRuntimeEventingConfiguration(eventURL string) (*RuntimeEventingConfiguration, error) {
+	validUrl, err := validateURL(eventURL)
+	if err != nil {
+		return nil, err
+	}
+
 	return &RuntimeEventingConfiguration{
 		EventingConfiguration: EventingConfiguration{
-			DefaultURL: url,
+			DefaultURL: validUrl,
 		},
-	}
+	}, nil
 }
 
+const ApplicationEventingURLScheme = "https://%s/%s/v1/events"
+
 type ApplicationEventingConfiguration struct {
 	EventingConfiguration
 }
 
-func NewApplicationEventingConfiguration(url string) *ApplicationEventingConfiguration {
+func NewApplicationEventingConfiguration(runtimeEventURL url.URL, appName string) (*ApplicationEventingConfiguration, error) {
+	var appEventURL string
+
+	if runtimeEventURL.Host == "" {
+		appEventURL = ""
+	} else {
+		appEventURL = fmt.Sprintf(ApplicationEventingURLScheme, runtimeEventURL.Host, appName)
+	}
+
+	validUrl, err := validateURL(appEventURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "while validating created application event url")
+	}
+
 	return &ApplicationEventingConfiguration{
 		EventingConfiguration: EventingConfiguration{
-			DefaultURL: url,
+			DefaultURL: validUrl,
 		},
+	}, nil
+}
+
+func NewEmptyApplicationEventingConfig() (*ApplicationEventingConfiguration, error) {
+	emptyURL, err := validateURL("")
+	if err != nil {
+		return nil, err
+	}
+
+	return &ApplicationEventingConfiguration{
+		EventingConfiguration: EventingConfiguration{
+			DefaultURL: emptyURL,
+		},
+	}, nil
+}
+
+func validateURL(rawURL string) (url.URL, error) {
+	validUrl, err := url.Parse(rawURL)
+	if err != nil {
+		return url.URL{}, err
+	}
+	if validUrl == nil {
```

Comment:
is there really a situation when error is nil and validURL is also nil?

Reply:
I look into the code deeper and there is no such situation.
I can remove this check.

Reply:
Alright, if validUrl can be nil, then keep it for safety reason.